cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/ReconcilerUtils.java

```diff
@@ -95,22 +99,68 @@ public static Future<Void> podsReady(Reconciliation reconciliation, PodOperator
+     * @return  Composite Future with the first result being the trust set containing the Cluster CA certificate and the second
+     *          result being the Cluster Operator public and private key to use for client authentication.
      */
-    public static CompositeFuture clientSecrets(Reconciliation reconciliation, SecretOperator secretOperator) {
+    public static CompositeFuture pemClientCertificates(Reconciliation reconciliation, SecretOperator secretOperator) {
         return Future.join(
-                getSecret(secretOperator, reconciliation.namespace(), KafkaResources.clusterCaCertificateSecretName(reconciliation.name())),
-                getSecret(secretOperator, reconciliation.namespace(), KafkaResources.secretName(reconciliation.name()))
+                pemTrustSet(reconciliation, secretOperator),
+                clientAuthIdentitySecret(reconciliation, secretOperator).compose(secret -> Future.succeededFuture(PemAuthIdentity.clusterOperator(secret)))
         );
     }
 
+    /**
+     * Utility method which helps to get the set of trusted certificates needed to bootstrap different clients used during
+     * the reconciliation.
+     *
+     * @param reconciliation Reconciliation Marker
+     * @param secretOperator Secret operator for working with Kubernetes Secrets that store certificates
+     *
+     * @return  Future containing the trust set to use for client authentication.
+     */
+    public static Future<PemTrustSet> pemTrustSet(Reconciliation reconciliation, SecretOperator secretOperator) {
+        return getSecret(secretOperator, reconciliation.namespace(), KafkaResources.clusterCaCertificateSecretName(reconciliation.name()))
+                .compose(secret -> Future.succeededFuture(new PemTrustSet(secret)));
+    }
+
+    /**
+     * Utility method which helps to get the set of client auth identities needed to bootstrap different clients used during
+     * the reconciliation.
+     *
+     * @param reconciliation Reconciliation Marker
+     * @param secretOperator Secret operator for working with Kubernetes Secrets that store certificates
+     *
+     * @return  Future containing the Kubernetes Secret with the Cluster Operator public and private key in both PEM and PKCS12 format
+     */
+    public static CompositeFuture clientAuthIdentity(Reconciliation reconciliation, SecretOperator secretOperator) {
+        Promise<PemAuthIdentity> pemAuthIdentity = Promise.promise();
+        Promise<ClusterOperatorPKCS12AuthIdentity> pkcs12AuthIdentity = Promise.promise();
+        clientAuthIdentitySecret(reconciliation, secretOperator).onSuccess(secret -> {
+            pemAuthIdentity.complete(PemAuthIdentity.clusterOperator(secret));
+            pkcs12AuthIdentity.complete(new ClusterOperatorPKCS12AuthIdentity(secret));
+        });
+        return Future.join(pemAuthIdentity.future(), pkcs12AuthIdentity.future());
```

Comment:
This segment seems quite confusing. I don't think they should return a CompositeFuture as that is a meaning less. Why not have some record for returning both? Or wh not use single class representing both the trustset and authentication identity? Are they really used separately anywhere?

Reply:
Sometimes we use PemAuthIdentity with ClusterOperatorPKCS12AuthIdentity and sometimes with PemTrustSet instead. As discussed elsewhere though I guess we might be able to remove ClusterOperatorPKCS12AuthIdentity if ZK client can be created with only PEM files

Reply:
> Sometimes we use PemAuthIdentity with ClusterOperatorPKCS12AuthIdentity and sometimes with PemTrustSet instead.

That seems weird -> PemAuthIdentity and ClusterOperatorPKCS12AuthIdentity both represent the user certificate, or?

I would expect you to use everywhere a pair:
* either PemTrustSet and ClusterOperatorPKCS12AuthIdentity
* or PemTrustSet and PemAuthIdentity

That sounds to me like you could merge the PemTrustSet into both of them (and for example have it null if not set in the user operator or something like that).

Reply:
It would simplify the methods with just one param etc.

Reply:
Yeah sorry, that's my mistype, it's as you stated:
- either PemTrustSet and ClusterOperatorPKCS12AuthIdentity
- or PemTrustSet and PemAuthIdentity

The other reason for having them separate is if in future we make it optional to have mutual TLS enabled for the operator connection to the Kafka brokers, we might want to only pass PemTrustSet and no AuthIdentity. But for how the code is today I like your suggestion of only passing the AuthIdentity class and having PemTrustSet be part of it, then we could always refactor to pass one or the other in future if we make that change. What do you think?

Reply:
Looking at the DefaultAdminClientProvider it makes sense there at least for them to be separate, so we can build the config accordingly, so I also wonder whether it is strange to have to pass to that method `(pemAuthIdentity.trustSet, pemAuthIdentity)`

Reply:
Well, I guess:
* You can always have one of them set to null (return null) even within a single class with different parts. If you today have everywhere the two parameters, you will anyway end up with one of them being null. So IMHO the question is more `Do you anywhere pass only one of PemTrustSet, PKCS12AuthIdentity or PemAuthIdentity?` If not, if you always pass one of the two pairings, then I think you do not need two parameters. It does not matter that in `DefaultAdminClientProvider` you sometimes expect one of those to be null as it will still be null whether as a direct parameter or in another class, or? 🤷
* I do not think we have any plans to make mTLS optional. The issue is more with making the whole TLS optional (as the whole TLS might happen on a different layer such as service mesh). But I think if we ever make TLS optional, we will likely replace it with some other kind of identity. So having a single class that in the future might be extended for other types of credentials IMHO plays well with that idea and will give us better results than adding more and more parameters to all the methods. E.g. if we use some `TlsPemIdentity` (and `TlsPKCS12Indetity`) that includes both the trust set and the client cert today, we can easily replace it in the future with some`AbstractIdentity` that might take the form of either the `TlsPemIdentity` or the `SpiffeIdentity` or whatever we use. 

In any case, even if you want to keep the parameters separate, the CompositeFuture is IMO not nice for a public method and it should use some record to wrap both of them at least in this place to pass the credentials out of this method.

Reply:
I've updated the methods in `ReconcilerUtils` to either return one of `PemTrustSet`, `ClusterOperatorPkcs12AuthIdentity` or `PemAuthIdentity`, or to return a new record that contains the two `AuthIdentity` classes.

Do you want me to also create records to be able to pass the `AuthIdentity` and `TrustSet` classes together rather than separately, or shall I leave them separate?

Reply:
Yeah, I think having a record for it would be useful since that will make it easier to refactor it in the future if needed to support more different credential types etc.